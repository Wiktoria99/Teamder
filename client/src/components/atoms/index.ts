export * from './BackButton';
export * from './CustomTextField';
export * from './CustomButton';
export * from './SideBarButton';
