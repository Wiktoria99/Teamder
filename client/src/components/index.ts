export * from "./atoms";
export * from "./atoms";
export * from "./molecules";
export * from "./organisms";
