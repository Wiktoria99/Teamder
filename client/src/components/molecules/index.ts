export * from './RegisterSteps';
export * from './TeamItem';
export * from './Suggestions';
export * from './Sidebar';
export * from './TeamList';
export * from './SuggestionItem';
export * from './SuggestionList';
export * from './NotificationItem';
export * from './NotificationList';
