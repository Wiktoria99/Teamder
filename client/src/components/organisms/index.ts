export * from './Layout';
export * from './FormWrapper';
export * from './MainWrapper';
export * from './BarWrapper';
