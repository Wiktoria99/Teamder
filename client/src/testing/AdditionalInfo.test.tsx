import { shallow, mount } from "enzyme";
import toJson from "enzyme-to-json";
import { AdditionalInfo, CustomTextField, CustomButton } from '@/components';
import { cleanup } from '@testing-library/react';

afterEach(cleanup);

const request = {
  user_name: 'test',
  password: 'test',
  password2: 'test',
  name: 'test',
  surname: 'test',
  email: 'test',
  location: 'test',
  age: 2,
  photo_src: 'test',
  bio: 'test',
  social_media_URL1: 'test',
  social_media_URL2: 'test',
  social_media_URL3: 'test',
  list_of_interests: ['test'],
};

it('AdditionalInfo renders without crashing', () => {
  shallow(<AdditionalInfo registerInfo={request} setRegisterInfo='' updateProgressState=''/>);
});

it('AdditionalInfo matches snapshot', () => {
  const tree = shallow(<AdditionalInfo registerInfo={request} setRegisterInfo='' updateProgressState=''/>);
  expect(toJson(tree)).toMatchSnapshot();
});

it("accepts AdditionalInfo props", () => {
  const wrapper = mount(<AdditionalInfo registerInfo={request} setRegisterInfo='' updateProgressState=''/>);
  expect(wrapper.props().registerInfo).toEqual(request);
  expect(wrapper.props().setRegisterInfo).toEqual('');
  expect(wrapper.props().updateProgressState).toEqual('');
});

it("find button and textfields", () => {
  const wrapper = mount(<AdditionalInfo registerInfo={request} setRegisterInfo='' updateProgressState=''/>);
  expect(wrapper.find(CustomTextField).length).toBe(5);
  expect(wrapper.contains(CustomButton)).toBe(true);
});
