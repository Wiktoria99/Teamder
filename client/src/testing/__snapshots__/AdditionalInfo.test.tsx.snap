// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AdditionalInfo matches snapshot 1`] = `
<Fragment>
  <form
    className="makeStyles-form-1"
    onSubmit={[Function]}
  >
    <WithStyles(WithStyles(ForwardRef(TextField)))
      color="secondary"
      label="Biogram"
      multiline={true}
      onChange={[Function]}
      value="test"
      variant="standard"
    />
    <WithStyles(WithStyles(ForwardRef(TextField)))
      color="secondary"
      label="Social Media URL #1"
      onChange={[Function]}
      value="test"
      variant="standard"
    />
    <WithStyles(WithStyles(ForwardRef(TextField)))
      color="secondary"
      label="Social Media URL #2"
      onChange={[Function]}
      value="test"
      variant="standard"
    />
    <WithStyles(WithStyles(ForwardRef(TextField)))
      color="secondary"
      label="Social Media URL #3"
      onChange={[Function]}
      value="test"
      variant="standard"
    />
    <WithStyles(WithStyles(ForwardRef(TextField)))
      color="secondary"
      label="Photo source file"
      onChange={[Function]}
      value="test"
      variant="standard"
    />
    <WithStyles(WithStyles(ForwardRef(Button)))
      className="makeStyles-button-2"
      color="secondary"
      type="submit"
      variant="contained"
    >
      <WithStyles(ForwardRef(Typography))
        variant="button"
      >
        Kontynuuj
      </WithStyles(ForwardRef(Typography))>
    </WithStyles(WithStyles(ForwardRef(Button)))>
  </form>
  <div
    className="makeStyles-formLinkDisclaimer-4"
    onClick={[Function]}
  >
    <span>
      Powr√≥t
    </span>
  </div>
</Fragment>
`;
